pkg/grpc/api.go

```diff
@@ -494,3 +503,12 @@ func (a *api) ExecuteStateTransaction(ctx context.Context, in *runtimev1pb.Execu
 	}
 	return &empty.Empty{}, nil
 }
+
+func (a *api) isSecretAllowed(storeName, key string) bool {
+	config, ok := a.secretsConfiguration[storeName]
+	if !ok {
+		// By default if a configuration is not defined for a secret store, return true.
+		return true
+	}
+	return config.IsSecretAllowed(key)
```

Comment:
I think this is a bit more natural, but not a big deal. You can do it either way. Thanks!

```suggestion
	if config, ok := a.secretsConfiguration[storeName]; ok {
		return config.IsSecretAllowed(key)
	}
	// By default if a configuration is not defined for a secret store, return true.
	return true
```